tests/test_algorithms.py

```diff
@@ -83,3 +83,27 @@ def test__split_tests_calculates_avg_test_duration_only_on_present_tests(self, a
         expected_first, expected_second = expected
         assert first.selected == expected_first
         assert second.selected == expected_second
+
+    def test_each_group_is_assigned_a_test_front_loaded(self):
+        item = namedtuple("item", "nodeid")
+
+        durations = {"a": 100, "b": 1, "c": 1, "d": 1, "e": 1, "f": 1, "g": 1, "h": 1}
+
+        items = [item(x) for x in ["a", "b", "c", "d", "e", "f", "g", "h"]]
+
+        groups = duration_based_chunks(8, items, durations)
+
+        for i in range(7):
+            assert groups[i].selected != []
+
+    def test_each_group_is_assigned_a_test_back_loaded(self):
+        item = namedtuple("item", "nodeid")
+
+        durations = {"a": 1, "b": 1, "c": 1, "d": 1, "e": 1, "f": 1, "g": 1, "h": 100}
+
+        items = [item(x) for x in ["a", "b", "c", "d", "e", "f", "g", "h"]]
+
+        groups = duration_based_chunks(8, items, durations)
+
+        for i in range(7):
+            assert groups[i].selected != []
```

Comment:
Is this same as `assert all(group.selected for group in groups)`?

Reply:
That looks like it would probably work; maybe just add a comment to indicate that we're making sure there aren't any empty groups.  Or maybe explicitly check that the size of each group is nonzero.

Reply:
>  we're making sure there aren't any empty groups ... check that the size of each group is nonzero

IMO `all(group.selected for group in groups)` does both and is explicit enough so no comments or further explanations would be needed. Just my opinion ofc 🙂